test/diff/idioms_pkgs_4/out-pure.nix

```
@@ -163,8 +163,7 @@ in
       cc =
         let
           nativePrefix =
-            {
-              # switch
+            {  # switch
```

A lot of these test cases have two spaces


```suggestion
            { # switch
```

---

It is introduced deliberately here in order to prevent idempotency issues in some very rare special cases. The test suite may not make it look like it but the entirety of Nixpkgs hits that edge case only once. Take a look at the input here:

```
nativePrefix = { # switch
  i686-solaris = "/usr/gnu";
  x86_64-solaris = "/opt/local/gcc47";
}.${system} or "/usr";
```

Naively, it would be formatted like this:

```
nativePrefix =
  { # switch
    i686-solaris = "/usr/gnu";
    x86_64-solaris = "/opt/local/gcc47";
  }
  .${system} or "/usr";
```

Because the comment and `i686-solaris` are vertically aligned, the comment will be associated with the `i686-solaris` and not with the `{`. This kind of respecting vertical alignment is necessary because too many people write code like:

```
foo = # comment
          and now the code
```

and without respecting that vertical alignment the comment would not be associated with its correct code. Anyways, because of this, a second formatting would wrongly associate the `switch` comment with `solaris` (which was not the intent of the original code), and thus produce 

```
nativePrefix =
  {
    # switch
    i686-solaris = "/usr/gnu";
    x86_64-solaris = "/opt/local/gcc47";
  }
  .${system} or "/usr";
```

which is different from the output after one formatting, and thus an idempotency issue. The "fix" is to insert one single space before comments that should not be aligned with the line that follows, so that they will be parsed correctly.

---

Knowing it is a pre-existing behaviour makes me feel better. Although I'm unsure how I feel about the ugly hack now being more exposed by this PR.

I guess, although it is covered a lot in our test suite, in reality it should be fairly rare to run into comments that need to be explicitly mis-aligned with the following value?